src/LRS.js

```diff
@@ -454,27 +455,35 @@ TinCan client library
             }
             if (typeof cfg.callback !== "undefined") {
                 callbackWrapper = function (err, xhr) {
-                    var result = xhr.responseText,
+                    var result,
                         boundary,
                         section,
+                        headers,
                         statement,
+                        attachmentMap = {},
                         i;
 
                     if (err === null) {
                         if (! cfg.params.attachments) {
-                            result = TinCan.Statement.fromJSON(result);
+                            result = TinCan.Statement.fromJSON(xhr.responseText);
                         }
                         else {
                             boundary = xhr.getResponseHeader("Content-Type").split("boundary=")[1];
-                            result = result.split("--" + boundary + "--")[0].split("--" + boundary);
+                            result = xhr.responseText.split("--" + boundary + "--")[0].split("--" + boundary);
 
                             statement = JSON.parse(result[1].split("\r\n\r\n")[1]);
 
                             for (i = 2; i < result.length; i += 1) {
                                 section = result[i].split("\r\n\r\n");
-                                statement = TinCan.LRS.prototype._assignAttachmentContent({statements: [statement]}, section[0].split("\r\n"), section[1].trim()).statements[0];
+                                headers = section[0].split("\r\n");
+                                for (i = 0; i < headers.length; i += 1) {
+                                    if (headers[i].indexOf("X-Experience-API-Hash") >= 0) {
+                                        attachmentMap[TinCan.Utils.getXAPIHashFromHeader(headers[i])] = section[1].replace(/^\s+|\s+$/g,"");
```

Comment:
This `.replace` is the reason I was concerned about the above as well, this will effectively trim valid whitespace from the body of the content when it shouldn't. We can't rely on the body of the message part not having leading/trailing whitespace, which is the point of the boundary conditions. We likely need to set up some more advanced test cases to make sure we not mangling the attachments.